### PR TITLE
Allocations tests for LSMR

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: src/lsmr.jl [hunk 1]
```diff
@@ -32,26 +32,26 @@ export lsmr
...
 
-  (A'A + λ² I) x = A'b
+  (AᵀA + λ² I) x = Aᵀb
 
 (and therefore to CRLS) but is more stable.
 
-LSMR produces monotonic residuals ‖r‖₂ and optimality residuals ‖A'r‖₂.
-rt is formally equivalent to CRLS, though can be slightly more accurate.
+LSMR produces monotonic residuals ‖r‖₂ and optimality residuals ‖Aᵀr‖₂.
+It is formally equivalent to CRLS, though can be slightly more accurate.
```
Comment (dpo) [inline src/lsmr.jl]: I think it would be fair to replace "slightly" with "substantially" here.

File: src/lsmr.jl [hunk 2]
```diff
@@ -194,8 +195,9 @@ function lsmr(A :: AbstractLinearOperator, b :: AbstractVector{T};
     err_vec[mod(iter, window) + 1] = ζ
     iter >= window && (err_lbnd = @knrm2(window, err_vec))
 
-    # Update h, h_har, x.
-    hbar = h - (θbar * ρ / (ρold * ρbarold)) * hbar
+    # Update h, hbar and x.
+    δ = θbar * ρ / (ρold * ρbarold) # δₖ = θbarₖ * ρₖ / (ρₖ₋₁ * ρbarₖ₋₁)
+    @kaxpby!(n, 1.0, h, -δ, hbar)   # ĥₖ = hₖ - δₖ * ĥₖ₋₁
```
Comment (dpo) [inline src/lsmr.jl]: In the comment, ĥₖ is a \hat{h}. Could you make it a \bar{h}?

Comment (dpo) [reply src/lsmr.jl]: Maybe in the code too?!

Comment (amontoison) [reply src/lsmr.jl]: The only \bar{h} available is this one `ħ`. But It's difficult to distinguish with `h`.

Comment (dpo) [reply src/lsmr.jl]: OK.

**Comment by dpo** [comment]
Thank you.

**Pull request merged**